CHANGELOG.md

```diff
@@ -0,0 +1,23 @@
+# Changelog
+
+## [1.1.1](https://github.com/uphold/anonymizer/releases/tag/v1.1.1) (2019-10-29)
+- Add lint script and apply lint fixes to codebase [\#19](https://github.com/uphold/anonymizer/pull/19) ([waldyrious](https://github.com/waldyrious))
+- Add uphold-scripts [\#17](https://github.com/uphold/anonymizer/pull/17) ([waldyrious](https://github.com/waldyrious))
+- Update . notation [\#16](https://github.com/uphold/anonymizer/pull/16) ([Americas](https://github.com/Americas))
+- Update package.json [\#15](https://github.com/uphold/anonymizer/pull/15) ([waldyrious](https://github.com/waldyrious))
+- Update README [\#13](https://github.com/uphold/anonymizer/pull/13) ([fixe](https://github.com/fixe))
+- Update dependencies [\#12](https://github.com/uphold/anonymizer/pull/12) ([fixe](https://github.com/fixe))
+- Update license [\#11](https://github.com/uphold/anonymizer/pull/11) ([fixe](https://github.com/fixe))
+- Update version to 1.1.1 [\#10](https://github.com/uphold/anonymizer/pull/10) ([fixe](https://github.com/fixe))
```

Comment:
Supposedly because it was created after the release tag, which is what we use for managing releases. Looks like an after-the-fact commit to fix something that was missing.

Reply:
That doesn't seem quite right:

```
❯ git show -s 5d7050e
5d7050e - (tag: 1.1.1) Update version to 1.1.1 (2 months ago) <Tiago Ribeiro>

❯ git tag --contains 5d7050e
1.1.1
```

But clearly something odd happened to the git graph — see 94b9a86 vs. f5954d0, which seem to be [unconnected](https://github.com/uphold/anonymizer/compare/94b9a86...f5954d0). With `diff -u <(git show 94b9a86) <(git show f5954d0)` you can see that the difference is that the latter includes the changes from #7, while the former doesn't (nor it shows

From what `git log --graph` shows me locally:

```
⋮
|  
* f5954d0 - (tag: 1.1.0) Initial commit (3 months ago) <Tiago Ribeiro>
* 9db9a42 - (tag: 1.0.3) Add support for circular references (3 months ago) <Tiago Ribeiro>
| *   8ed3865 - (tag: 1.0.2) Merge pull request #7 from uphold/support/handle-circular-references (3 months ago) <Francisco Cardoso>
| |\  
|/ /  
| * 69940e6 - Add support for circular references (3 months ago) <Tiago Ribeiro>
|/  
* 94b9a86 - (tag: 1.0.1) Initial commit (3 months ago) <Tiago Ribeiro>
```

...I'd guess that #7 was merged with a merge commit (8ed3865) instead of via a rebase, and a new commit (9db9a42) was pushed to fix that; but somehow that the git graph got messed up in the process. (Note that only f5954d0 onwards appears in GitHub's [network graph](https://github.com/uphold/anonymizer/network) and in the [list of commits](https://github.com/uphold/anonymizer/commits/master).)

Regardless, maybe github-changelog-generator could somehow detect that 5d7050e is contained in the tag `1.1.1`, as shown above. But I suppose this may be an edge case. For now, I'll manually edit the entry to be in the correct section; let's see what happens when we cut a new release.

Reply:
The changelog generator always rewrites the entire file, so manual edits would always need to be reapplied.

If this repo has some issues with its git history, maybe we should postpone adding the changelog generator to the mix. Maybe discuss with @fixe the current release process to see how we can use it or adapt it.